store/index/gc.go

```diff
@@ -0,0 +1,179 @@
+package index
+
+import (
+	"bufio"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var log = logging.Logger("storethehash/index")
+
+// gcInterval is how often to run garbage collection.
+const gcInterval = 30 * time.Minute
+
+// garbageCollector is a goroutine that runs periodically to search for and
+// remove stale index files.
+func (i *Index) garbageCollector() {
+	defer close(i.gcDone)
+
+	var gcDone chan struct{}
+	hasUpdate := true
+
+	// Run 1st GC 1 minute adter startup.
+	t := time.NewTimer(time.Minute)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for {
+		select {
+		case _, ok := <-i.updateSig:
+			if !ok {
+				// Channel closed; shutting down.
+				cancel()
+				if gcDone != nil {
+					<-gcDone
+				}
+				return
+			}
+			hasUpdate = true
+		case <-t.C:
+			if !hasUpdate {
+				// Nothing new, keep waiting.
+				t.Reset(gcInterval)
+				continue
+			}
+
+			gcDone = make(chan struct{})
+			go func() {
+				defer close(gcDone)
+				start := time.Now()
+				if err := i.gc(ctx); err != nil {
```

Comment:
put a debug log to indicate that GC is starting?

Reply:
Done